docs/bsp.md

```diff
@@ -1207,6 +1208,198 @@ Clients can use this information to reconstruct the tree of sbt meta builds. The
 `parent` information can be defined from `children` but it's provided by the server to
 simplify the data processing on the client side.
 
+## BSP Connection Protocol
+
+The Build Server Protocol defines a standard convention for clients to
+connect with BSP servers. This protocol has been designed such that:
+
+1. Clients do not require beforehand knowledge about a specific server
+   implementation to be able to connect to it.
+1. Clients do not need to know the specifics of what kind of connections are
+   supported by concrete build tools.
+1. Clients can connect to build tools installed both at the machine and at the
+   workspace level.
+1. Multiple versions of the same build tool can have different connection
+   mechanisms.
+1. Multiple different BSP servers can run in the same workspace directory.
+
+The following protocol to discover and connect to BSP servers is inspired by
+[Jupyter's
+Kernelspec](https://jupyter-client.readthedocs.io/en/stable/kernels.html#kernelspecs),
+even though it has some notorious differences that are explained below.
```

Comment:
I'm not sure we should remove the adjective, I want to stress that the differences are fundamental rather than tweaks. Do you want to propose another adjective instead?
  
Maybe the following formulation would be better?
  
> The following connection protocol is inspired by [Jupyter's Kernelspec](https://jupyter-client.readthedocs.io/en/stable/kernels.html#kernelspecs) and improves on some of its aspects to abide by the protocol requirements stated above.

Reply:
Your reformulation is fine. "notable" or "noteworthy" could work as well. "notorious differences" is not a positive thing

> famous or well known, typically for some bad quality or deed.

Reply:
Oh thanks! I didn't know it had a negative connotation.